File: 021-Retirement_py/retirement.py
def abstract_cal(initial_savings, info):

    initial_savings = initial_savings + initial_savings * info["rate_of_return"] + info["contribution"] 
    return initial_savings


def retirement(start_age, initial_savings, working_info, retired_info):
    """
    Prints the current status of an individual's retirement account.

    The dictionaries both have these fields:
        "months","contribution","rate_of_return"

    Args:
        start_age (int): At what age (in months) does the individual start
        initial_savings (float): initial savings in dollars
        working_info (dict): information about working
        retired_into (dict): information about retirement
    Returns:
        None
    """
    for index in range(working_info["months"]):
        total_months = start_age + index
        yr = total_months // 12
        m = total_months % 12
        print("Age {:3d} month {:2d} you have ${:,.2f}".format(yr, m, initial_savings))
        initial_savings = abstract_cal(initial_savings,working_info)


    for index2 in range(retired_info["months"]):
        total_months = start_age + working_info["months"] + index2
        yr = total_months // 12
        m = total_months % 12
        print("Age {:3d} month {:2d} you have ${:,.2f}".format(yr, m, initial_savings))
        initial_savings = abstract_cal(initial_savings, retired_info)

def main():
    start_age = 327
    initial_savings = 21345
    working_info={
    "months":489,
    "contribution":1000.0,
    "rate_of_return": 0.045/12
    }

    retired_info={
    "months":384,
    "contribution":-4000.0,
    "rate_of_return": 0.01/12
    }
    retirement(start_age, initial_savings, working_info, retired_info)

main()
        

File: 051-Retirement_cpp/retirement.cpp
#include "std_lib_facilities.h"
#include <iomanip>
#include<iostream>

class RetireInfo {
    public:
    int months;
    double contribution;
    double rate_of_return;
};

void printLine(int current_age, double balance){
    cout << "Age " << setw(3) << (current_age/12) << " month " << setw(2) << (current_age % 12) << " you have $" << fixed << setprecision(2) << balance << "\n";
}

double updateBalance(double balance, RetireInfo info){
    return balance * (info.rate_of_return) + balance + (info.contribution);
}

void retirement (int startAge, double initial, RetireInfo working, RetireInfo retired){
    for (int i = 0; i < working.months;i++){
        int current_age = startAge + i;
        printLine(current_age, initial);
        initial = updateBalance(initial, working);
    }

    for (int j =0; j<retired.months;j++){
        int current_age = startAge + working.months+ j;
        printLine(current_age, initial);
        initial =updateBalance(initial, retired);
    }

}

int main(void){
    RetireInfo working;
    working.months = 489;
    working.contribution = 1000;
    working.rate_of_return = 0.045/12;

    RetireInfo retired;
    retired.months = 384;
    retired.contribution = -4000;
    retired.rate_of_return = 0.01/12;

    retirement (327,21345, working, retired);
    return EXIT_SUCCESS;
}